hooks/viewer/useViewer.js

```diff
@@ -25,7 +28,27 @@ export default function useViewer() {
   }, [accessToken, viewer]);
 
   useEffect(() => {
-    if (viewer) setAccount(viewer);
+    if (viewer) {
+      setAccount(viewer);
+
+      // identify the user
+      const isAuthenticated = viewer?._id;
```

Comment:
Thinking out loud here. From a platform consistency perspective does it make sense to introduce optional chaining into the codebase right now when [it requires](https://node.green/#ES2020-features-optional-chaining-operator-----) Node 14.5+ to use? The dev docker image is currently using Node v12, so I'm curious if there are any environment changes required or if recent versions of Next.js just handle it like magic.

Reply:
Yes @balibebas, Nextjs has option chaining built-in since the beginning of this year, since some 9.x version.
The code is transpiled anyway, the docker node image doesn't play a big role here, some even older version would probably work aswell, official nextjs requirement is 10.13.

I like using optional chaining in more cases where it is possibly, the reduce some verbosity, especially regarding data-fetching.